processors/record_linker/processor.py

```diff
@@ -0,0 +1,72 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import uuid
+import logging
+from .. import base
+logger = logging.getLogger(__name__)
+
+
+# Module API
+
+def process(conf, conn):
+
+    # Trial m2m
+    TRIAL_TABLES = [
+        'trials_conditions',
+        'trials_interventions',
+        'trials_locations',
+        'trials_organisations',
+        'trials_persons',
+        'trials_publications',
+    ]
+
+    # Unlink wrong linked records
+    # [will be reprocessed on next processors run]
+    count = 0
+    for record in base.helpers.iter_rows(conn, 'database', 'records', orderby='id'):
+        # Already unlinked
+        if not record['trial_id']:
+            continue
+        trial_records = list(conn['database']['records'].find(trial_id=record['trial_id']))
```

Comment:
I haven't tested, but maybe it's possible to add `conn['database']['records'].columns.id != record['id']` as another param to the `find()` call, so you won't need to bother removing the current record from the array.